docs/writing-docs/doc-blocks.md

```diff
@@ -0,0 +1,331 @@
+TODO: waiting to be vetted
+</div>
+
+In DocsPage, a component’s description is shown at the top of the page. For [supported frameworks], the component description is automatically extracted from a docgen component above the component in its source code. It can also be set by the `docs.description` parameter.
+
+```js
+
+export default {
+  title: 'CustomDescription'
+  parameters: {
+    docs: { 
+        description: { 
+            component: 'some component _markdown_' 
+        } 
+    },
+  }
+};
+
+export const WithStoryDescription = Template.bind({});
+WithStoryDescription.parameters = {
+  docs: { 
+      description: { 
+          story: 'some story **markdown**' 
+        } 
+    },
+};
+```
+
+#### MDX
+
+In MDX, the `Description` shows the component’s description using the same heuristics as the DocsPage. It also accepts a markdown parameter to show any user-provided Markdown string.
+
+```js
+
+import { Description } from '@storybook/addon-docs/blocks';
+import dedent from 'ts-dedent';
+import { Button } from './Button';
+
+<Description of={Button} />
+<Description markdown={dedent`
+  ## Custom description
+
+  Insert fancy markdown here.
+`}/>
+```
+
+### Story
+
+Stories (component examples) are the basic building blocks in Storybook. In Storybook Docs, stories are rendered in the `Story` block.
+
+![Docs blocks with stories](./docblock-story.png)
+
+#### DocsPage
+
+In DocsPage a `Story` block is generated for each story in your [CSF] file, wrapped with a `Preview` wrapper that gives it a toolbar on top (in the case of the first “primary” story) and a source code preview underneath.
+
+#### MDX
+
+In MDX, the `Story` block is not only a way of displaying stories, but also the primary way to define them. Storybook looks for `Story` instances with the `name` prop, either defined at the top level of the document, or directly beneath a [Preview](#preview) block defined at the top level:
+
+```js
+import { Story } from '@storybook/addon-docs/blocks';
+import { Button } from './Button';
+
+export const Template = (args) => <Button {...args} />;
+
+<Story name="Basic" args={{ label: ‘hello’ }}>
+  {Template.bind({})
+</Story>
+```
+
+You can also reference existing stories in Storybook by ID:
+
+```js
+
+import { Story } from '@storybook/addon-docs/blocks';
+
+<Story id="some-component--some-name" />
+```
+
+#### Inline rendering
+
+<div style="background-color:#F8FAFC">
+TODO: ask tom/michael about the preview iframe. as we've been calling it canvas so far and now move to preview.
+And also inline rendering support location
+</div>
+
+In Storybook’s Canvas, all stories are rendered in the [Preview iframe] for isolated development. In Storybook Docs, when [inline rendering is supported by your framework](locate-link), inline rendering is used by default for performance and convenience. However, you can force iframe rendering with `docs: { inlineStories: false }` parameter, or `inline={false}` in MDX.
```

Comment:
Add link to supported framework page

Reply:
I've left that one, as i was unable to find the specific documentation for this. If anyone can point me in the right direction i would appreciate it.